lsp-mode.el

```diff
@@ -167,13 +167,21 @@ the buffer when it becomes large."
   :type 'boolean
   :package-version '(lsp-mode . "6.1"))
 
-(defcustom lsp-enable-semantic-highlighting nil
+(defcustom lsp-semantic-highlighting nil
   "Enable/disable semantic highlighting as proposed at
  https://github.com/microsoft/vscode-languageserver-node/pull/367.
 This feature is not yet part of the official LSP spec and may
 occasionally break as language servers are updated."
   :group 'lsp-mode
-  :type 'boolean)
+  :type '(choice
+          (const :tag "Disable" nil)
+          (const :tag "Immediate" :immediate)
+          (const :tag "Deferred" :deferred)))
+
+(defcustom lsp-semantic-highlighting-window-size 70
```

Comment:
I'd rename this to `lsp-semantic-highlighting-context-lines`, following a similar naming convention as occur's `list-matching-lines-default-context-lines` or compile's `compilation-context-lines`.

Reply:
thanks for testing this! I've tried to address these changes.
Following a proposal by @yyoncho, I also added a commit that uses font-lock callbacks to refontify changed or newly visited buffer parts. Doesn't work very well yet though (triggering refontification on buffer content changes doesn't really seem like a good idea to me as long as we don't use the new semantic tokenization protocol, as the first fontification run may often precede the arrival of the new highlighting information we need).
Maybe I should just get rid of the last commit, merge the non-font-lock state of this PR and revisit font-lock integration as soon as support for the new highlighting protocol gets implemented?

Reply:
> Maybe I should just get rid of the last commit, merge the non-font-lock state of this PR and revisit font-lock integration as soon as support for the new highlighting protocol gets implemented?

Make sense to me.

Reply:
I now changed back to a non-font-lock implementation. I suggest to give this a few more days of testing, then I'd squash all commits except for the `inhibit-field-motion` thing and merge this PR